builtin/providers/aws/resource_aws_route53_record.go

```diff
@@ -407,12 +407,13 @@ func resourceAwsRoute53RecordBuildSet(d *schema.ResourceData, zoneName string) (
 		rec.HealthCheckId = aws.String(v.(string))
 	}
 
-	if v, ok := d.GetOk("weight"); ok {
-		rec.Weight = aws.Int64(int64(v.(int)))
-	}
+	//if v, ok := d.GetOk("weight"); ok {
+	//	rec.Weight = aws.Int64(int64(v.(int)))
+	//}
 
 	if v, ok := d.GetOk("set_identifier"); ok {
 		rec.SetIdentifier = aws.String(v.(string))
+		rec.Weight = aws.Int64(int64(d.Get("weight").(int)))
```

Comment:
@apparentlymart See the original issue and reason for this PR: https://github.com/hashicorp/terraform/issues/3189

It was being set _in most cases_, not for zero.